ledger/eval/eval.go

```diff
@@ -1358,8 +1430,43 @@ func (eval *BlockEvaluator) endOfBlock() error {
 		}
 	}
 
-	err = eval.state.CalculateTotals()
-	if err != nil {
+	// Try to pay the proposer.
+	{
+		proposer := eval.block.Proposer()
+		payout := eval.block.ProposerPayout()
+		// The proposer won't be present yet when generating a block
+		if !proposer.IsZero() {
+			if !payout.IsZero() {
+				err := eval.state.Move(eval.block.FeeSink, proposer, payout, nil, nil)
```

Comment:
why not checking proposer closed account before the Move and do not pay if closed?

Reply:
This makes them ineligible.
https://github.com/jannotti/go-algorand/blob/absenteeism/ledger/ledgercore/validatedBlock.go#L86-L93

But let's leave this open until I convince myself that it can't be "spoofed".

Reply:
okay, payout will be zero in this case and `Move` is not invoked. Do we have test for this? I skimmed eval_test and eval_simple_test and did not find.

Reply:
A go e2e test. mining_test.go